tfhe/src/shortint/oprf.rs

```diff
@@ -0,0 +1,191 @@
+use crate::shortint::server_key::LookupTableOwned;
+use crate::shortint::ServerKey;
+use concrete_csprng::seeders::Seed;
+
+impl ServerKey {
+    fn create_random_from_seed(&self, seed: Seed) -> Ciphertext {
+        let mut ct = self.create_trivial(0);
+
+        let mut generator = RandomGenerator::<ActivatedRandomGenerator>::new(seed);
+
+        for mask_e in ct.ct.get_mut_mask().as_mut() {
+            *mask_e = generator.random_uniform::<u64>();
+        }
+
+        ct
+    }
+
+    /// Uniformly generates a random value in `[0, 2^random_bits_count[`
+    /// `2^random_bits_count` must be smaller than the message modulus
+    pub fn generate_oblivious_pseudo_random(
+        &self,
+        seed: Seed,
+        random_bits_count: u64,
+    ) -> Ciphertext {
+        assert!(1 << random_bits_count < self.message_modulus.0);
+
+        self.generate_oblivious_pseudo_random_message_and_carry(seed, random_bits_count)
+    }
+
+    /// Uniformly generates a random value in `[0, 2^random_bits_count[`
+    pub fn generate_oblivious_pseudo_random_message_and_carry(
+        &self,
+        seed: Seed,
+        random_bits_count: u64,
+    ) -> Ciphertext {
+        assert!(self.message_modulus.0.is_power_of_two());
+        let message_bits_count = self.message_modulus.0.ilog2() as u64;
+
+        assert!(self.carry_modulus.0.is_power_of_two());
+        let carry_bits_count = self.carry_modulus.0.ilog2() as u64;
+
+        assert!(random_bits_count < carry_bits_count + message_bits_count);
+
+        self.generate_oblivious_pseudo_random_custom_encoding(
+            seed,
+            random_bits_count,
+            1 + carry_bits_count + message_bits_count,
+        )
+    }
+
+    /// Uniformly generates a random value in `[0, 2^random_bits_count[`
+    /// The output in in the form 0000rrr000noise (rbc=3, fbc=7)
+    pub(crate) fn generate_oblivious_pseudo_random_custom_encoding(
+        &self,
+        seed: Seed,
+        random_bits_count: u64,
+        full_bits_count: u64,
+    ) -> Ciphertext {
+        let ct = self.create_random_from_seed(seed);
+
+        let p = 1 << random_bits_count;
+
+        let delta = 1_u64 << (64 - full_bits_count);
+
+        let poly_delta = 2 * self.bootstrapping_key.polynomial_size().0 as u64 / p;
+
+        let acc: LookupTableOwned =
+            self.generate_lookup_table_no_encode(|x| (2 * (x / poly_delta) + 1) * delta / 2);
+
+        let mut ct = self.apply_lookup_table(&ct, &acc);
+
+        lwe_ciphertext_plaintext_add_assign(&mut ct.ct, Plaintext((p - 1) * delta / 2));
+
+        ct.degree = Degree::new(p as usize - 1);
+
+        ct
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use crate::shortint::{Ciphertext, ServerKey};
+    use concrete_csprng::seeders::Seed;
+    use rayon::prelude::*;
+    use statrs::distribution::ContinuousCDF;
+    use std::collections::HashMap;
+
+    fn square(a: f64) -> f64 {
+        a * a
+    }
+
+    #[test]
+    fn oprf_test_uniformity() {
```

Comment:
you may need to add the `_ci_run_filter` suffix to make sure the test runs because of our regex for parameterized tests

so the test would be named `oprf_test_uniformity_ci_run_filter`

Reply:
Done